src/api.py

```diff
@@ -0,0 +1,44 @@
+import hashlib
+import requests
+
+class Response:
+    def __init__(self, filename, type_of_data, size, url, checksum):
+        self.filename = filename
+        self.type_of_data = type_of_data
+        self.size=size
+        self.url=url
+        self.checksum=checksum
+
+
+def generate_filename(url):
+    """
+    split url on basis of '/' and start iterating from end till you find non empty value
+    """
+    for i in range(len(url.split('/'))-1):
+        name = url.split('/')[-(i+1)]
+        if not(name =='' ):
+            break
+    return name
+
+
+def fetch(url,location):
+    # Fetching data from URL
+    try:
```

Comment:
Please do not warp your code in try/except at this stage:
1. we never or super rarely catch and ignore exceptions as you are doing here
2. we surely rarely if ever swallow them and print a message instead.

Instead let the exceptions bubble up as they should